Uses itk to perform a gaussian blur

salamb opened the pull request.

Commit:
Uses itk to perform a gaussian blur

Commit:
Show how to install libraries and add modules

Installs a minimal version of ITK and adds a
gaussian blur module

cdeepakroy commented on Dockerfile:
```diff
@@ -0,0 +1,88 @@
+#cmake release v3.6.0
+RUN git checkout e31084e65745f9dd422c6aff0a2ed4ada6918805 && \
+./bootstrap --system-curl && make && make install
+#by default cmake does not use system ssl
```
You can combine all commands to install cmake as follows

``` Dockerfile
RUN cd / && git clone https://github.com/Kitware/CMake.git && cd CMake && \
    git checkout e31084e65745f9dd422c6aff0a2ed4ada6918805 && \
    ./bootstrap --system-curl && make && make install
```

That way all commands to install cmake will be in one cached docker layer.

Also test, if the [apt-get install](https://github.com/InsightSoftwareConsortium/ITKIOOpenSlide/blob/master/test/Docker/Dockerfile#L7) works for cmake. Alternatively, i think you can download the latest pre-complied version for linux using wget from [here](https://cmake.org/download/) and install it.

cdeepakroy commented on Dockerfile:
```diff
@@ -0,0 +1,88 @@
+RUN mkdir ninja
+WORKDIR ninja
+
+RUN wget https://github.com/ninja-build/ninja/releases/download/v1.7.1/ninja-linux.zip && \
```
You can combine all the above commands into one as follows i think:

``` Dockerfile
RUN cd / && mkdir -p ninja && cd ninja && \
    wget https://github.com/ninja-build/ninja/releases/download/v1.7.1/ninja-linux.zip && \
```

cdeepakroy replied:
Also by looking at [this](https://github.com/InsightSoftwareConsortium/ITKIOOpenSlide/blob/master/test/Docker/Dockerfile#L18) it seems like ninja can be apt-get installed as well.

cdeepakroy replied:
I think it is because he is using a different base image. So wget install is fine for now.

cdeepakroy commented on Dockerfile:
```diff
@@ -0,0 +1,88 @@
+RUN mkdir ITKbuild
+WORKDIR ITKbuild
+
+RUN cmake \
```
All the above commands for cloning and checking out ITK can be combined with this RUN command as follows:

``` Dockerfile
RUN cd / && git clone https://github.com/InsightSoftwareConsortium/ITK.git && cd ITK && \
    git checkout $(git describe --abbrev=0 --tags) && \
    cd ../ && mkdir -p ITKbuild && cd ITKbuild \
    cmake \
```

Commit:
Updated to use dsarchive/histomicstk:v0.1.3

Assumes that the image dsarchive/histomicstk:v0.1.4 contains
an installation of python and itk
